### PR TITLE
Ignoring src in media tags

### PR DESCRIPTION
**Pull request by AbdelrhmanMagdy**
Ignoring big base64 encoded src for video and audio html tags to prevent large payload tokens and memory leaks.
Addressing [this issue](https://github.com/microsoft/clarity/issues/407#issuecomment-1992055905)

### CODE REVIEW

File: packages/clarity-js/src/layout/node.ts [hunk 1]
```diff
@@ -182,6 +182,16 @@ export default function (node: Node, source: Source): Node {
                     let linkData = { tag, attributes };
                     dom[call](node, parent, linkData, source);
                     break;
+                case "VIDEO":
+                case "AUDIO":
+                case "SOURCE":
+                    // Ignoring any base64 src attribute for media elements to prevent big unused tokens to be sent and shock the network 
+                    if(Constant.Src in attributes && attributes[Constant.Src].startsWith("data:")){
```
Comment (ender336) [inline packages/clarity-js/src/layout/node.ts:189]: nit: missing a space after if

**Review by ender336 — APPROVED** [review]

File: packages/clarity-js/src/layout/node.ts [hunk 2]
```diff
@@ -182,6 +182,16 @@ export default function (node: Node, source: Source): Node {
                     let linkData = { tag, attributes };
                     dom[call](node, parent, linkData, source);
                     break;
+                case "VIDEO":
+                case "AUDIO":
+                case "SOURCE":
+                    // Ignoring any base64 src attribute for media elements to prevent big unused tokens to be sent and shock the network 
+                    if (Constant.Src in attributes && attributes[Constant.Src].startsWith("data:")) {
+                        delete attributes[Constant.Src]
```
Comment (swaathee) [inline packages/clarity-js/src/layout/node.ts:190]: do mediaTag.attributes[Constant.Src] = empty string instead of deleting here

Comment (swaathee) [reply packages/clarity-js/src/layout/node.ts:190]: also how is empty src handled on visualiaze?

Comment (AbdelrhmanMagdy) [reply packages/clarity-js/src/layout/node.ts:190]: An empty "src" attribute results in displaying an HTML media element, such as audio/video, with a playback length of 0.00.

**Pull request merged**